src/corefx/System.Globalization.Native/localeNumberData.cpp

```diff
@@ -31,9 +47,259 @@ enum LocaleNumberData : int32_t
 
 /*
 Function:
+NormalizePattern
+
+Returns a numeric string pattern in a format that we can match against the appropriate managed pattern.
+*/
+UnicodeString* NormalizePattern(const UnicodeString *srcPattern, UnicodeString *destPattern, bool isNegative)
+{
+	// A srcPattern example: "#,##0.00 C;(#,##0.00 C)" but where C is the international currency symbol (chCurrencySign)
+	// The positive pattern comes first, then an optional negative pattern separated by a semicolon
+	// A destPattern example: "(C n)" where C represents the currency symbol, and n is the number
+	destPattern->remove();
+
+	int iStart = 0;
+	int iEnd = srcPattern->length() - 1;
+	int32_t iNegative = srcPattern->indexOf(chPatternSeparator);
+	if (iNegative >= 0)
+	{
+		if (isNegative)
+		{
+			iStart = iNegative + 1;
+		}
+		else
+		{
+			iEnd = iNegative - 1;
+		}
+	}
+
+	bool minusAdded = false;
+	bool digitAdded = false;
+	bool currencyAdded = false;
+	bool spaceAdded = false;
+
+	for (int i = iStart; i <= iEnd; i++)
+	{
+		UChar32 ch = srcPattern->char32At(i);
```

Comment:
We are mixing codepoints UChar32 and codeunits (UChar) here.  Can we just use 'UChar' here instead?